sources/bloodhound/src/output.rs

```diff
@@ -0,0 +1,58 @@
+use std::io::{Error, Write};
+
+use crate::results::ReportResults;
+
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+pub trait ReportWriter {
+    fn write(&self, report: &ReportResults, output: &mut dyn Write) -> Result<(), Error>;
+}
+
+pub struct TextReportWriter {}
+
+impl ReportWriter for TextReportWriter {
+    /// Writes a text formatted report to the provided output destination.
+    fn write(&self, report: &ReportResults, output: &mut dyn Write) -> Result<(), Error> {
+        if let Some(name) = &report.metadata.name {
+            writeln!(output, "{:17}{}", "Benchmark name:", name)?;
```

Comment:
There isn't a great way to just drop in a constant in the formatting string template. I would need to do something like:

```rust
writeln!(output, "{:pad$}{}", "Benchmark name:", name, pad=PADDING)
```

That seems less readable to me, especially for something that probably isn't going to ever change.

I can make that change if you think it would be better, but my personal inclination would be to leave it as is. Just let me know your preference.

Reply:
If you don't think it will ever change then I'm fine to leave it. If we do ever find it changing that might be the time to switch to the `pad=PADDING` method.

Reply:
Definitely - sounds like a good plan!